### PR TITLE
[JUDGE] added savedHistory

### PR DESCRIPTION
**Pull request by akowal3**
# Summary

Added `client.savedHistory` to obtain meaningful information about other clients - resources (actual and reported), allocation & taxation (expected and actual).
Added `judge.saveHistoryInfo` which is run in InspectHistory to take the values, pre-process them and save them to our client

## Test Plan

- Added some tests

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func createSOMASServer(
 				shared.Speaker:   0,
 			},
 			SpeakerID:          shared.Team1,
-			JudgeID:            shared.Team2,
+			JudgeID:            shared.Team4,
```
Comment (tobhil98) [inline internal/server/server.go]: Nooo... Make sure you don't change this.

File: internal/clients/team4/judge.go [hunk 2]
```diff
@@ -33,16 +35,66 @@ type judge struct {
...
+	Taxation   valuePair // amount of tax paid and expected
+	Allocation valuePair // amount of allocation taken and granted
+	Lied       int       // number of times the island has lied
+}
+
+func (j *judge) saveHistoryInfo(iigoHistory *[]shared.Accountability, lieCounts *map[shared.ClientID]int, turn uint) {
+	accountabilityMap := map[shared.ClientID][]rules.VariableValuePair{}
+	for _, acc := range *iigoHistory {
+		client := acc.ClientID
+		if accountabilityMap[client] != nil {
```
Comment (tobhil98) [inline internal/clients/team4/judge.go]: Any chance there is a cleaner way of doing this? Can we initialise the list with all clientID's maybe?

**Review by tobhil98 — CHANGES_REQUESTED** [review]
Looks good. Make sure to revert server.go

Also, this looks to be branched off client-Judge, so you should either create a PR back to client-Judge, or more preferably at this point, wait for a PR from client-Judge to custom-client to be merged first.

File: internal/server/server.go (hunk 1 above)
Comment (tobhil98) [reply internal/server/server.go]: @Spruudis We should probably request to have this changed to some random assignment. It's a pretty big advantage to start with a role.

**Comment by tobhil98** [comment]
To be merged after #96

File: internal/server/server.go (hunk 1 above)
Comment (akowal3) [reply internal/server/server.go]: can't we only revert it when creating PR to the main branch? it's just a waste of time and effort (I know it's a small one) to revert this, create local PR and then change it back again for testing. 
if you have a very strong opinion about it, then I'll do it 😉

Comment (tobhil98) [reply internal/server/server.go]: I do kinda have a strong opinion about it :P
Firstly this is only used when testing the Judge, so there is no use in having it on the client. Secondly, if we start checking in code which we shouldn't touch we might forget and either assume an outcome that is not the case or forget to fix it later

File: internal/clients/team4/judge.go (hunk 2 above)
Comment (akowal3) [reply internal/clients/team4/judge.go]: done

File: internal/clients/team4/judge.go [hunk 3]
```diff
@@ -49,13 +49,13 @@ type judgeHistoryInfo struct {
 
 func (j *judge) saveHistoryInfo(iigoHistory *[]shared.Accountability, lieCounts *map[shared.ClientID]int, turn uint) {
 	accountabilityMap := map[shared.ClientID][]rules.VariableValuePair{}
+	for _, clientID := range shared.TeamIDs {
+		accountabilityMap[clientID] = []rules.VariableValuePair{}
+	}
+
```
Comment (tobhil98) [inline internal/clients/team4/judge.go]: Nice!

**Comment by akowal3** [comment]
#96 was merged. I resolved the conflicts (ie. accepted my versions 🤣 )

File: internal/server/server.go (hunk 1 above)
Comment (akowal3) [reply internal/server/server.go]: ok, I'll fix it

**Review by tobhil98 — APPROVED** [review]

**Pull request merged**